dump/mapping/prepbufr_obs_builder.py

```diff
@@ -23,14 +24,16 @@ def __init__(self, mapping_path, log_name=os.path.basename(__file__)):
     def _get_reference_time(self, input_path) -> np.datetime64:
         path_components = Path(input_path).parts
 
-        dump = re.compile(r'\w+\.(?P<year>\d{4})(?P<month>\d{2})(?P<day>\d{2})')
-        test = re.compile(r'(?P<year>\d{4})(?P<month>\d{2})(?P<day>\d{2})(?P<hour>\d{2})')
-
-        for idx, component in enumerate(reversed(path_components[:-1])):
-            dump_match = dump.match(component)
-            test_match = test.match(component)
+        dump_regex = r'\w+\.(?P<year>\d{4})(?P<month>\d{2})(?P<day>\d{2})'
+        test_regex = r'(?P<year>\d{4})(?P<month>\d{2})(?P<day>\d{2})(?P<hour>\d{2})'
 
+        for idx, component in enumerate(reversed(path_components)):
+            dump_match = re.match(dump_regex, component)
+            test_match = re.match(test_regex, component)
             if dump_match:
```

Comment:
recompiling the regular expression in each iteration of the loop which is expensive, but maybe don't care here